src/DataflowPass.cpp

```diff
@@ -221,32 +233,36 @@ void DataflowPass::analyzeCFG(CFG *cfg, ProgramFunction &preProgramFunction,
     std::string currentBranch = cfg->getBranchName();
 
     if (currentBranch == FIRST_BRANCH_NAME) {
-        preProgramFunction.addProgramPoint(ProgramPoint(currentBranch));
+        preProgramFunction.addProgramBlock(ProgramBlock(currentBranch));
         llvm::SetVector<Instruction *> instructions = cfg->getInstructions();
 
-        ProgramPoint postProgramPoint =
-            this->programFunction.getProgramPoint(currentBranch, true);
+        ProgramBlock postProgramBlock =
+            this->programFunction.getProgramBlock(currentBranch, true);
 
         auto fnName = this->programFunction.getFunctionName();;
         auto f_iterator = this->F->args();
+        ProgramPoint *p = new ProgramPoint(0);
+        postProgramBlock.add(p);
 
         for(auto a : this->annotations.getAllParameterAnnotationsWithoutFields(fnName)) {
             if(a->getAnnotationType() == AnnotationType::MustCallAnnotation) {
                 int index = a->getParameterIndex();
                 auto f_iterator = this->F->args();
                 auto arg = std::next(f_iterator.begin(), index);
                 ProgramVariable var = ProgramVariable(arg);
-                auto pvas = postProgramPoint.getPVASRef(var, true);
+                auto pvas = postProgramBlock.getPoint(0)->getPVASRef(var, true);
```

Comment:
Are we sure that program blocks are guaranteed to have at least one point? There's nothing in the program block documentation that says that we should assume that invariant.

I can imagine a situation in the future where we might want empty blocks. For example, in Java we sometimes used empty blocks to represent the implicit `finally` LUB after a `try` and a `catch`. So, we should either document the invariant that program blocks always have at least one point, or we shouldn't assume it here. I'd probably prefer the former (i.e., document the invariant), but it's up to you.

Reply:
(Especially given how often you make the assumption, enforcing the invariant is almost certainly a better design.)